Arm/Syntax.lean

```diff
@@ -6,11 +6,35 @@ Author(s): Siddharth Bhat
 Provide convenient syntax for writing down state manipulation in Arm programs.
 -/
 import Arm.State
+import Arm.Memory.Separate
 
 namespace ArmStateNotation
 
 /-! We build a notation for `read_mem_bytes $n $base $s` as `$s[$base, $n]` -/
 @[inherit_doc read_mem_bytes]
 syntax:max term noWs "[" withoutPosition(term)  ","  withoutPosition(term) noWs "]" : term
 macro_rules | `($s[$base,$n]) => `(read_mem_bytes $n $base $s)
+
+
+/-! Notation to specify the frame condition for non-memory state components. E.g.,
+`REGS_UNCHANGED_EXCEPT [[.GPR 0, .PC]] (sf, s0)` is sugar for
```

Comment:
Do we want the double paren?

```
`REGS_UNCHANGED_EXCEPT [[.GPR 0, .PC]] 
```

At least for me, I find:

```
`REGS_UNCHANGED_EXCEPT [.GPR 0, .PC] 
```

easier to read.

Reply:
Tell me how to remove the extra `[]`, and I'll do it :)

Reply:
Doing the naive thing (i.e., removing the block parens. around `$regs`) in the syntax and macro rules isn't cutting it